### PR TITLE
`/link` enhancements

### PR DESCRIPTION
**Pull request by DonaldKellett**
- Use link description as key
- Make reply ephemeral when no target user is specified

### CODE REVIEW

File: src/commands/link.ts [hunk 1]
```diff
@@ -61,7 +49,7 @@ export const data = new SlashCommandBuilder()
       .setName("topic")
       .setDescription("The topic to link to")
       .setRequired(true)
-      .addChoices(Object.keys(LINKS).map((k) => [k, k]))
+      .addChoices(Object.keys(LINKS).map((k) => [LINKS[k].description, k]))
```
Comment (kazk) [inline src/commands/link.ts:52]: Looks like the key is used to select.
So `/link topic:kata` shows two entries for example, and it's a bit annoying.

Maybe use a unique prefix + description as key?

```typescript
.addChoices(Object.keys(LINKS).map((k) => [`${k}: ${LINKS[k].description}`, k]))
```

Comment (kazk) [reply src/commands/link.ts:52]: Wait, that might not help.

File: src/commands/link.ts [hunk 2]
```diff
@@ -74,5 +62,8 @@ export const call = async (interaction: CommandInteraction) => {
   const linkInfo = LINKS[topic];
   let replyMsg = `See ${hyperlink(linkInfo.description, hideLinkEmbed(linkInfo.url))}`;
   if (target) replyMsg = `${userMention(target.id)} ${replyMsg}`;
-  await interaction.reply(replyMsg);
+  await interaction.reply({
+    content: replyMsg,
+    ephemeral: !target,
```
Comment (kazk) [inline src/commands/link.ts:67]: I missed this. What's the reason behind this change?

Is it common to use `/link` for themselves? I guess we can allow opting out of posting like `/rankup`.

Comment (DonaldKellett) [reply src/commands/link.ts:67]: My rationale was that `/link` would usually be used to direct a user to some part of the docs (or some other link), in which case the `target` option should be specified in order to mention the specified user. But in the rare case where a target user is not specified, the caller probably just wanted to read up on the topic themselves so there is little reason to make the response visible to everyone.

Comment (kazk) [reply src/commands/link.ts:67]: Even without being mentioned, the target user will most likely see it because the message will be shown right below their message.

I don't think mentioning the target is critical:
```text
no0bUser99: Hey guys I want to create my first kata, how do I do this?
(/) Kacarott used `/link`
Codewars:   See https://docs.codewars.com/authoring
no0bUser99: Ok thanks!
```

```text
no0bUser99: Hey guys I want to create my first kata, how do I do this?
(/) Kacarott used `/link`
Codewars:   @no0bUser99 See https://docs.codewars.com/authoring
no0bUser99: Ok thanks!
```

> But in the rare case where a target user is not specified, the caller probably just wanted to read up on the topic themselves

I disagree with this assumption. The target is optional, so it's easy to miss. I think it happens more accidentally.
Using `?link` in a reply was also opt-in.

Comment (DonaldKellett) [reply src/commands/link.ts:67]: > Even without being mentioned, the target user will most likely see it because the message will be shown right below their message.
> 
> I don't think mentioning the target is critical:
> 
> ```
> no0bUser99: Hey guys I want to create my first kata, how do I do this?
> (/) Kacarott used `/link`
> Codewars:   See https://docs.codewars.com/authoring
> no0bUser99: Ok thanks!
> ```
> 
> ```
> no0bUser99: Hey guys I want to create my first kata, how do I do this?
> (/) Kacarott used `/link`
> Codewars:   @no0bUser99 See https://docs.codewars.com/authoring
> no0bUser99: Ok thanks!
> ```

That's assuming that the reply always comes immediately after the query which is not always the case. Sometimes, the query may be ignored for a while by other users and additional, unrelated discussion may have occurred before someone decides to respond to the original query, in which case the `/link` invocation would appear out of place unless the caller also explicitly mentions the target immediately afterwards:

```
(e.g. in #beginners )
n00bUser1: Hey guys I want to create my first Kata, how do I do this?
n00bUser2: Hi all, I think there is a syntax error in my code. How do I fix this?

console.log Hello World!
1337User1: n00bUser2 please go to help-solve and follow the guidelines
randomUser1: ...
...
1337User2: /link authoring
codewars: See https://docs.codewars.com/authoring/kata
n00bUser2: ???
(at this point n00bUser1 may have left and would not notice the bot reply the next time they log in)
```

> I disagree with this assumption. The target is optional, so it's easy to miss. I think it happens more accidentally.
Using ?link in a reply was also opt-in.

Fair point. I think we can adjust the command if many users find the current behavior problematic.
